pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,723 @@
+	"os"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/rancher/aks-operator/controller"
+	aksv1 "github.com/rancher/aks-operator/pkg/apis/aks.cattle.io/v1"
+	"github.com/rancher/norman/condition"
+	apimgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	apiprojv3 "github.com/rancher/rancher/pkg/apis/project.cattle.io/v3"
+	utils2 "github.com/rancher/rancher/pkg/app"
+	"github.com/rancher/rancher/pkg/catalog/manager"
+	"github.com/rancher/rancher/pkg/controllers/management/clusterupstreamrefresher"
+	"github.com/rancher/rancher/pkg/controllers/management/rbac"
+	"github.com/rancher/rancher/pkg/dialer"
+	v3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	corev1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	mgmtv3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	projectv3 "github.com/rancher/rancher/pkg/generated/norman/project.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/kontainer-engine/drivers/util"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/project"
+	"github.com/rancher/rancher/pkg/ref"
+	"github.com/rancher/rancher/pkg/systemaccount"
+	"github.com/rancher/rancher/pkg/types/config"
+	typesDialer "github.com/rancher/rancher/pkg/types/config/dialer"
+	"github.com/rancher/rancher/pkg/wrangler"
+	wranglerv1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	systemNS            = "cattle-system"
+	aksAPIGroup         = "aks.cattle.io"
+	aksV1               = "aks.cattle.io/v1"
+	aksOperatorTemplate = "system-library-rancher-aks-operator"
+	aksOperator         = "rancher-aks-operator"
+	localCluster        = "local"
+	enqueueTime         = time.Second * 5
+)
+
+type aksOperatorValues struct {
+	HTTPProxy  string `json:"httpProxy,omitempty"`
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+	NoProxy    string `json:"noProxy,omitempty"`
+}
+
+type aksOperatorController struct {
+	clusterEnqueueAfter  func(name string, duration time.Duration)
+	secretsCache         wranglerv1.SecretCache
+	templateCache        v3.CatalogTemplateCache
+	projectCache         v3.ProjectCache
+	appLister            projectv3.AppLister
+	appClient            projectv3.AppInterface
+	nsClient             corev1.NamespaceInterface
+	clusterClient        v3.ClusterClient
+	catalogManager       manager.CatalogManager
+	systemAccountManager *systemaccount.Manager
+	dynamicClient        dynamic.NamespaceableResourceInterface
+	clientDialer         typesDialer.Factory
```

Comment:
It appears you are not using this although I would assume you would need to assign it the the rest config you create.

Reply:
I will check this